Bottom: 8b2394eab45312f4f421983062769f05b862b863
Top:    735990405a0f1750693a6ca14c8bef10bf094c55
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-28 00:50:38 +0100

Fix things a little - to merge below as appropriate - might need a revert in git HEAD


Revert "hid/gtk: Add hooks to tell the render when we start/stop drawing"

This reverts commit 332cb8ddc1cc032ac29c7d2b9a0051ccc228b6da.


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 27d80f7..94bf6f5 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -985,17 +985,6 @@ ghid_drawing_area_configure_hook (GHidPort *port)
     }
 }
 
-gboolean
-ghid_start_drawing (GHidPort *port)
-{
-  return TRUE;
-}
-
-void
-ghid_end_drawing (GHidPort *port)
-{
-}
-
 void
 ghid_screen_update (void)
 {
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index d7e406b..ab212b9 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -667,12 +667,6 @@ ghid_port_window_motion_cb (GtkWidget * widget,
 
   gdk_event_request_motions (ev);
 
-#if 0
-  if (!ghid_start_drawing (out))
-    return FALSE;
-#endif
-
-
   if (out->panning)
     {
       dx = gport->zoom * (x_prev - ev->x);
@@ -690,14 +684,8 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   queue_tooltip_update (out);
 #endif
 
-//  ghid_show_crosshair (FALSE);
-//  ghid_show_crosshair (TRUE);
-  if (moved) // && have_crosshair_attachments ())
-    ghid_draw_area_update (gport, NULL);
-
-#if 0
-  ghid_end_drawing (out);
-#endif
+  ghid_show_crosshair (FALSE);
+  ghid_show_crosshair (TRUE);
 
   return FALSE;
 }
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 3adf0f4..2061e40 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -499,8 +499,6 @@ void ghid_show_crosshair (gboolean show);
 void ghid_init_renderer (int *, char ***, GHidPort *);
 void ghid_init_drawing_widget (GtkWidget *widget, GHidPort *);
 void ghid_drawing_area_configure_hook (GHidPort *port);
-gboolean ghid_start_drawing (GHidPort *port);
-void ghid_end_drawing (GHidPort *port);
 void ghid_screen_update (void);
 gboolean ghid_drawing_area_expose_cb (GtkWidget *, GdkEventExpose *,
                                       GHidPort *);
